orttraining/orttraining/core/graph/gradient_builder.cc

```diff
@@ -921,35 +921,39 @@ IMPLEMENT_GRADIENT_BUILDER(GetFastGeluGradient) {
 }
 
 IMPLEMENT_GRADIENT_BUILDER(GetSendGradient) {
-  NodeAttributes new_attrs;
-  const auto& old_attrs = SrcNodeAttributes();
- 
-  new_attrs["src"] = MakeAttribute("src", old_attrs.at("dst").i());
-  new_attrs["dst"] = MakeAttribute("dst", old_attrs.at("src").i());
-  new_attrs["tag"] = MakeAttribute("tag", old_attrs.at("tag").i());  // keep tag the same for now
-  new_attrs["element_type"] = MakeAttribute("element_type", old_attrs.at("element_type").i());
+  // Send inputs: signal A, remote, data; outputs: signal B
+  // Recv inputs: signal B, remote; outputs: signal A', data'
+
+  std::vector<ArgDef> out_args;
+  out_args.push_back(GI(0));  // Signal
+  for (int i = 2; i < GetSrcNodeInputSize(); ++i) {
+    out_args.push_back(GI(i));  // Data
+  }
 
   return std::vector<NodeDef>{
       NodeDef("Recv",
-              {O(0)},
-              {GI(0), GI(1)},
-              new_attrs)};
+              {O(0), I(1)},  // {Signal, Remote}
```

Comment:
Humm. Does it mean if Send has dst=10, this Recv gets src=10? Looks like need need both of dst and src in Send and Recv input lists.

Reply:
I think this holds. Because it is backward right?
src of send is its own rank.  dst of recv is also its own rank.
So you flip, and then it still holds.

Reply:
I got your point now. We send tensor to dst=10 and receive tensor from src=10. Makes sense.